test/system/releaseTest.js

```diff
@@ -0,0 +1,43 @@
+const utils = require('../../src/utils')
+const fs = require('fs')
+
+describe('release', function () {
+  let env = null
+
+  before(function () {
+    sandbox.stub(fs, 'stat').yields(null, { size: 666 })
+    sandbox.stub(fs, 'writeFile').yields(null, 'ok')
+    sandbox.stub(fs, 'readFile').yields(null, 'ok')
+    sandbox.stub(console, 'info')
+    sandbox.spy(console, 'warn')
+    sandbox.stub(utils, 'exec').yields()
+    utils.exec.withArgs('git rev-parse --abbrev-ref HEAD').yields(null, 'master')
+    sandbox.stub(utils, 'labelPullRequestWithMetricMovement').resolves()
+    env = JSON.parse(JSON.stringify(process.env))
+  })
+
+  after(function () {
+    process.env = JSON.parse(JSON.stringify(env))
+    sandbox.restore()
+  })
+
+  describe('when all is ok', function () {
+    before(function (done) {
+      require('../../nodeApps/release')
+      setTimeout(done, 1000)
```

Comment:
Why the timeout?

Reply:
there are async things in there, so allow it some time to run

Reply:
If i call this file with childProcess.exec or childProcess.exec then I can give a real callback, but then I cannot stub the methods inside - I don't want the test to really create files or anything. If I rearrange the file so that I can test it better than this then I'm not testing what the file actually does. This works ok like this can I keep it?

Reply:
I have added comments to the code explaining why the timeout is necessary